### PR TITLE
mgr/rook: removing all the code related to OSDs creation/removal

### PR DESCRIPTION
**Pull request by rkachach**
Fixes: https://tracker.ceph.com/issues/64211

Related issues: https://github.com/rook/rook/issues/13600

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by rkachach** [comment]
jenkins retest this please

**Comment by rkachach** [comment]
jenkins test rook e2e

**Comment by rkachach** [comment]
jenkins test rook e2e

File: src/pybind/mgr/rook/module.py [hunk 1]
```diff
@@ -634,134 +609,11 @@ def apply_nfs(self, spec):
...
-
     def add_host_label(self, host: str, label: str) -> OrchResult[str]:
         return self.rook_cluster.add_host_label(host, label)
-    
+
     def remove_host_label(self, host: str, label: str, force: bool = False) -> OrchResult[str]:
         return self.rook_cluster.remove_host_label(host, label)
-    """
-    @handle_orch_error
-    def create_osds(self, drive_group):
```
Comment (jmolmo) [inline src/pybind/mgr/rook/module.py:698]: This is what is called in the orchestrators when we "apply" a drive_group. I would like not remove this method, i would suggest to raise an exception to inform the user that osd creation using rook orchestrator is not available for the moment.

File: src/pybind/mgr/rook/rook_cluster.py [hunk 2]
```diff
@@ -1204,48 +885,11 @@ def _update_mon_count(current, new):
...
-        else:
-            creator = DefaultCreator(inventory, self.coreV1_api, self.storage_class_name)
-        return self._patch(
-            ccl.CephCluster,
-            'cephclusters',
-            self.rook_env.cluster_name,
-            creator.add_osds(self.rook_pods, drive_group, matching_hosts)
-        )
-
-    def remove_osds(self, osd_ids: List[str], replace: bool, force: bool, mon_command: Callable) -> str:
```
Comment (jmolmo) [inline src/pybind/mgr/rook/rook_cluster.py:1228]: Same than with OSD creation. I would suggest to raise an exception. (This method is part of the orchestrators standard interface)

**Review by jmolmo — CHANGES_REQUESTED** [review]
Great thing.. IMHO Drive groups never were a good idea. 
The way we managed osds in the rook orchestrator goes totally against the k8s declarative nature.
I think now it is demonstrated very clearly that trying to translate "imperative" into "declarative" does not work, and requires lot of effort.

I think this is ok, but we need to keep the orchestrators interface "working", so please raise exceptions reporting the situation in the orchestrator standar methods to add/remove osds.

Note: Any idea about how we are going to provide OSD management using the Rook Orchestrator?

**Review by jmolmo — APPROVED** [review]

**Comment by rkachach** [comment]
jenkins test rook e2e

**Comment by rkachach** [comment]
jenkins test rook e2e

**Comment by rkachach** [comment]
jenkins test rook e2e

**Comment by rkachach** [comment]
jenkins test rook e2e

**Comment by rkachach** [comment]
> Great thing.. IMHO Drive groups never were a good idea. The way we managed osds in the rook orchestrator goes totally against the k8s declarative nature. I think now it is demonstrated very clearly that trying to translate "imperative" into "declarative" does not work, and requires lot of effort.
> 
> I think this is ok, but we need to keep the orchestrators interface "working", so please raise exceptions reporting the situation in the orchestrator standar methods to add/remove osds.
> 
> Note: Any idea about how we are going to provide OSD management using the Rook Orchestrator?

Not sure if this answers your question or not but at this moment there're no plans to have OSD mgmt being performed from the rook orchestrator. The idea is to have all OSD related operations being managed by the Rook operator by using the Yaml spec files.

**Pull request merged**